underscore.js

```diff
@@ -136,15 +136,35 @@
     };
   };
 
+  // Helper to determine if index is within the bounds of an array.
+  var isIndex = function(index, length) {
+    return index >= 0 && index < length;
+  };
+
   // Helper for collection methods to determine whether a collection
-  // should be iterated as an array or as an object
+  // should be iterated as an array or as an object.
   // Related: http://people.mozilla.org/~jorendorff/es6-draft.html#sec-tolength
   // Avoids a very nasty iOS 8 JIT bug on ARM-64. #2094
   var MAX_ARRAY_INDEX = Math.pow(2, 53) - 1;
   var getLength = property('length');
   var isArrayLike = function(collection) {
     var length = getLength(collection);
-    return typeof length == 'number' && length >= 0 && length <= MAX_ARRAY_INDEX;
+    return typeof length == 'number' && isIndex(length, MAX_ARRAY_INDEX);
+  };
+
+  // Helper to determine if the arguments to a function are from
+  // an iteration of `each`, `map`, and friends.
+  var isIterationCall = function(value, index, collection) {
```

Comment:
That'd be nifty, but I think it'd slow down common usage (it'll require an `#apply`). And I don't think the other contributors want more wrappers.